ios/brave-ios/Sources/PlaylistUI/PlayerModel.swift

```diff
@@ -86,13 +87,52 @@ public final class PlayerModel: ObservableObject {
 
   var currentTime: TimeInterval {
     guard let item = player.currentItem else { return 0 }
-    return max(0, min(duration, item.currentTime().seconds))
+    switch duration {
+    case .seconds(let timeInterval):
+      return max(0, min(timeInterval, item.currentTime().seconds))
+    case .unknown, .indefinite:
+      let currentTime = item.currentTime()
+      if currentTime.isIndefinite || currentTime.seconds.isNaN {
+        return 0
+      }
+      return currentTime.seconds
+    }
   }
 
-  var duration: TimeInterval {
-    guard let item = player.currentItem else { return 0 }
-    let seconds = item.asset.duration.seconds
-    return seconds.isNaN ? 0.0 : seconds
+  enum ItemDuration {
+    case unknown
+    case seconds(TimeInterval)
+    case indefinite  // Live video
+
+    var seconds: TimeInterval? {
+      if case .seconds(let timeInterval) = self {
+        return timeInterval
+      }
+      return nil
+    }
+
+    var isIndefinite: Bool {
+      if case .indefinite = self {
+        return true
+      }
+      return false
+    }
+
+    init(_ seconds: TimeInterval) {
+      if !seconds.isFinite || seconds == .greatestFiniteMagnitude {
+        self = .indefinite
+      } else {
+        self = .seconds(seconds)
+      }
+    }
+  }
+
+  var duration: ItemDuration {
+    guard let duration = player.currentItem?.asset.duration else { return .unknown }
+    if duration.isIndefinite {
```

Comment:
Might be good to check `isValid` here as well?

Reply:
Added check above this and return `.unknown` duration for invalid, tested with live streams which still get valid indefinite durations